lib/components/UploadPicker.vue

```diff
@@ -240,54 +300,116 @@ export default Vue.extend({
 		/**
 		 * Trigger file picker
+		 * @param folder Upload folders
 		 */
-		onClick() {
-			this.$refs.input.click()
+		onTriggerPick(folder = false) {
+			const input = this.$refs.input as HTMLInputElement
+			// Setup directory picking if enabled
+			if (this.canUploadFolders) {
+				input.webkitdirectory = !!folder
+			}
+			// Trigger click on the input to open the file picker
+			this.$nextTick(() => input.click())
 		},
 
 		/**
-		 * Start uploading
+		 * Helper for backwards compatibility that queries the content of the current directory
+		 * @param path The current path
 		 */
-		async onPick() {
-			let files = [...this.$refs.input.files] as File[]
-
-			// Detect and resolve conflicts
-			if (hasConflict(files, this.content)) {
-				// List of incoming files that are in conflict
-				const conflicts = files.filter((file: File) => {
-					return this.content.find((node: Node) => node.basename === file.name)
-				}).filter(Boolean) as File[]
-
-				// List of incoming files that are NOT in conflict
-				const uploads = files.filter((file: File) => {
-					return !conflicts.includes(file)
-				})
-
-				try {
-					// Let the user choose what to do with the conflicting files
-					const { selected, renamed } = await openConflictPicker(this.destination.basename, conflicts, this.content)
-					files = [...uploads, ...selected, ...renamed]
-				} catch (error) {
-					// User cancelled
-					showError(t('Upload cancelled'))
-					return
+		async getContent(path?: string) {
+			return Array.isArray(this.content) ? this.content : await this.content(path)
+		},
+
+		/**
+		 * Show a dialog to let the user decide how to proceed with invalid filenames.
+		 * The returned promise resolves to true if the file should be renamed and resolves to false to skip it the file.
+		 * The promise rejects when the user want to abort the operation.
+		 *
+		 * @param filename The invalid file name
+		 */
+		async showInvalidFileNameDialog(filename: string) {
+			return new PCancelable(async (resolve, reject) => {
+				await new DialogBuilder()
+					.setName(t('Invalid file name'))
+					.setSeverity('error')
+					.setText(t('"{filename}" contains invalid characters, how do you want to continue?', { filename }))
+					.setButtons([
+						{
+							label: t('Rename'),
+							type: 'primary',
+							callback: () => resolve(true),
+						},
+						{
+							label: t('Skip'),
+							callback: () => resolve(false),
+						},
+						{
+							label: t('Cancel'),
+							type: 'error',
+							callback: reject,
+						},
+					])
+					.build()
+					.show()
+			})
+		},
+
+		async handleConflicts(nodes: Array<File|Directory>, path: string): Promise<Array<File|Directory>|false> {
+			const invalidReplacement = ['-', '_', ' '].filter((c) => !this.forbiddenCharacters.includes(c))[0] ?? 'x'
+
+			try {
+				const content = path === '' ? this.currentContent : await this.getContent(path).catch(() => [])
+				const conflicts = getConflicts(nodes, content)
```

Comment:
Does `getCOnflicts` also check conflicts on folders and does it recursively? Or maybe the nodes list is flat?

Reply:
Actually what happens when the user upload a folder with a name that is already taken by another folder?
Is the new folder name updated? Or is the new folder merged into the existing one?
If the last option is used, it means that we have to recursively get the folder content to check for conflicts?

Reply:
The conflicts are resolved recursively like your file manager does on your OS.
`getConflicts` only gets conflicts in the current level, then it lets you decide what todo:

- Select the new one (aka "write into")
- Select the old one (aka skip the upload)
- Select both: Rename the new one